packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4210,6 +4180,9 @@ export class ContainerRuntime
 				const idAllocationBatchMessage: BatchMessage = {
 					contents: serializeOpContents(idAllocationMessage),
 					referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
+					// Note: For now, we will never stage ID Allocation messages.
+					// They won't contain personal info and no harm in extra allocations in case of discarding the staged changes
+					staged: false,
```

Comment:
i'm ok with this, but there is likely a small problem here with read write transition, and editing the file without content change. this probably exists outside staging mode. for instance, it we use id for data stores, we are in read, and we create a detached datastore fluid will transtion to wrtie, and edit the file, even if the the detached datastore is never attached.

Reply:
Yeah I'll be opening a task to figure out ID Compressor.